server/index.js

```diff
@@ -54,7 +54,27 @@ export default (cb) => {
       }
     }
   });
-  server.register([ goodPlugin, Inert, Blipp, jwt, hapiSwaggerPlugin, ...plugins ], (err) => {
+
+  const myPlugin = {
+    register: function (server, options, next) {
+      server.ext('onRequest', function (request, reply) {
+        // do something
+        if (request.headers['x-forwarded-proto'] ) {
+          request.headers['x-forwarded-proto'] = request.headers['x-forwarded-proto'].split(',').shift();
+        }
+        return reply.continue();
+      });
+      next();
+    }
+  };
+
+  myPlugin.register.attributes = {
+    name: 'myPlugin',
+    version: '1.0.0'
+  };
+
+
+  server.register([ myPlugin, goodPlugin, Inert, Blipp, jwt, hapiSwaggerPlugin, ...plugins ], (err) => {
```

Comment:
Since logic is already before initializing the server I don't think the plugin is needed anymore